### PR TITLE
chore(updatecli): preparatory improvement of end dates to avoid manifest duplication

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR adds a "concern" field to the end_dates moved from locals.yaml to updatecli values as a preparatory improvement allowing to use a single updatecli manifest to update all end dates.

I migrated end dates from locals.yaml to updatecli/values.yaml to avoid having to surcharge updatecli shared pipeline library function parameter (like https://github.com/jenkins-infra/azure/pull/740/files#diff-630b6e89e0ee5cbf4a232ba05a74890122a08ffb1e06cba69a165cded772bd1eR1), as forgetting to pass multiple values when executing updatecli locally is a common source of error.

Creating this separate PR so updatecli can have these changes in the main branches to properly validate https://github.com/jenkins-infra/azure/pull/740

Follow-up of:
- #734
- #737

### CODE REVIEW

**Review by dduportal — CHANGES_REQUESTED** [review]
There seems to be an inversion of fields:

```
-/+ resource "azuread_application_password" "fileshare_serviceprincipal_writer" {
      ~ application_object_id = "d69cd926-8dac-491e-8e4f-994fbb9765be" -> (known after apply)
      ~ end_date              = "2024-09-19T23:00:00Z" -> "2023-09-19T23:00:00Z" # forces replacement
      ~ id                    = "d69cd926-8dac-491e-8e4f-994fbb9765be/password/0ef6782d-7154-462b-8629-81541d93cc02" -> (known after apply)
      ~ key_id                = "0ef6782d-7154-462b-8629-81541d93cc02" -> (known after apply)
      ~ start_date            = "2024-06-19T15:32:10Z" -> (known after apply)
      ~ value                 = (sensitive value)
        # (2 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

File: updatecli/values.yaml [hunk 1]
```diff
@@ -5,3 +5,18 @@ github:
...
+    end_date: "2024-06-20T23:00:00Z"
+    concern: "contributors.jenkins.io"
+  infraci_docs_jenkins_io_fileshare_serviceprincipal_writer:
+    end_date: "2024-08-07T23:00:00Z"
+    concern: "docs.jenkins.io"
+  infraci_pluginsjenkinsio_fileshare_serviceprincipal_writer:
+    end_date: "2024-07-27T00:00:00Z"
+    concern: "plugins.jenkins.io"
+  infraci_stats_jenkins_io_fileshare_serviceprincipal_writer:
+    end_date: "2023-09-19T23:00:00Z"
```
Comment (dduportal) [inline updatecli/values.yaml:21]: Found the inversion

```suggestion
    end_date: "2024-09-19T23:00:00Z"
```

Comment (dduportal) [reply updatecli/values.yaml:21]: Ref. https://github.com/jenkins-infra/azure/pull/741/files#diff-1a27c7a4b6293913f23d9922f0f57cb36a737ae6d88cdbbaa49a463b2a2cc360L7

Comment (lemeurherve) [reply updatecli/values.yaml:21]: Damn, that was the change I made to test the manifest, commited by mistake 😅

**Review by dduportal — APPROVED** [review]
- Terraform says: `No changes. Your infrastructure matches the configuration.` => your change is 👍 
- Updatecli diff shows 3 `[checkIfExpirySoonExpired]` conditions (not expired yet): LGTM

**Pull request merged**